### ISSUE TITLE
Cannot open filenames with special characters on Windows

### ISSUE THREAD

**Issue by chris0552** [keepassxreboot/keepassxc#8751]
## Overview
When i try to open a Keepass-file containing special characters (like ä, ß, ...) from Windows-explorer, it does not work if KeepassXC is already opened. Also opening an URL to another Keepass-file does not work if special charactrs are in it's filename.

## Steps to Reproduce
1. Install KeePassXC on Windows
2. open KeepassXC
3. double-click a Keepass-file from Windows-explorer

## Expected Behavior
Keepass should show up with the Keepass-file opened.

## Actual Behavior
Keepass shows up but the file is not opened.

## Environment
KeePassXC - 2.7.4
Revision: 63b2394

Operating System: Windows 10

**Comment by jNullj** [comment]
Using:
KeePassXC - Version 2.7.1
Revision: 5916a8f
on win10

Can not replicate this issue, created a file named 1ß.kdbx
when i double click it keepass tries to open it:
![image](https://user-images.githubusercontent.com/15849761/200678681-447a3c52-18c6-46a1-8ae9-fd8038287710.png)

**Comment by jNullj** [comment]
Tried updating to
KeePassXC - Version 2.7.4
Revision: 63b2394
win10

Can replicate the issue

**Comment by jNullj** [comment]
this commit seems to touch the related area and it is also exactly made between those two versions
https://github.com/keepassxreboot/keepassxc/commit/dd15db721a3d8d9e276e6363ae5d51e35c33e4c7

**Comment by jNullj** [comment]
After a bit of reading i think there is an issue there

In main.cpp the `FindFirstFile` function is being used (line 125):
```
hFind = FindFirstFile(fileInfo.absoluteFilePath().toUtf8(), &findFileData);
```

While microsoft docs state that:
"In the ANSI version of this function, the name is limited to MAX_PATH characters. To extend this limit to approximately 32,000 wide characters, call the Unicode version of the function (FindFirstFileExW), and prepend "\\?\" to the path. For more information, see [Naming a File](https://learn.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file)."

My bet is that using `FindFirstFileExW` should fix this issue.

I would love to make those changes and send a pull request...

Read more here: [microsoft fileapi.h docs](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findfirstfilea)

**Comment by droidmonkey** [comment]
Easy fix and I agree that is the cause

**Comment by jNullj** [comment]
I was a bit wrong, the function used is A variant insted of W, adding explicit W and fixing casting issues solves this, but i also went and tested issue #7139 which started this little adventure and while it does not change the filename capitals it adds the extention to the name...

**Comment by jNullj** [comment]
Ok i got it working, solved both this and #7139
